pennylane_lightning/core/src/simulators/lightning_kokkos/catalyst/CMakeLists.txt

```diff
@@ -0,0 +1,92 @@
+cmake_minimum_required(VERSION 3.20)
+
+project(lightning_kokkos_catalyst LANGUAGES CXX)
+
+set(CATALYST_FILES LightningKokkosSimulator.cpp CACHE INTERNAL "" FORCE)
+add_library(lightning_kokkos_catalyst SHARED ${CATALYST_FILES})
+
+include(FetchContent)
+
+if(CATALYST_SRC_PATH)
+    if(NOT IS_ABSOLUTE ${CATALYST_SRC_PATH})
+        message(FATAL_ERROR " CATALYST_SRC_PATH=${CATALYST_SRC_PATH} must be set to an absolute path")
+    endif()
+    if(CATALYST_GIT_TAG)
+        message(WARN " Setting `CATALYST_SRC_PATH=${CATALYST_SRC_PATH}` overrides `CATALYST_GIT_TAG=${CATALYST_GIT_TAG}`")
+    endif()
+
+    # Acquire local git hash and use for CATALYST_GIT_TAG
+    execute_process(COMMAND git rev-parse --short HEAD
+        WORKING_DIRECTORY ${CATALYST_SRC_PATH}
+        OUTPUT_VARIABLE CATALYST_GIT_TAG
+    )
+    message(INFO " Building against local Catalyst - path: ${CATALYST_SRC_PATH} - GIT TAG: ${CATALYST_GIT_TAG}")
+
+    target_include_directories(lightning_kokkos_catalyst PUBLIC ${CATALYST_SRC_PATH}/runtime/lib/backend/common)
+    target_include_directories(lightning_kokkos_catalyst PUBLIC ${CATALYST_SRC_PATH}/runtime/include)
```

Comment:
Do you mean like against the full `catalyst_runtime` project?
As defined [here](https://github.com/PennyLaneAI/catalyst/blob/main/runtime/CMakeLists.txt)?
Because doing that will add a lot of unnecessary compilation.

We don't have any of these headers as part of an interface library (for example) that I could link against.

Reply:
Or do you mean using the `catalyst_qir_runtime` target?

Reply:
Because this guy has some other dependencies.

Reply:
I think we'll need to define some upstream changes to the catalyst runtime then to better allow this. Ideally, defining an INTERFACE target allows us to pull in the headers, without any expectation of building the source material. I'd say this can be tracked for a follow-up work, as it will more easily allow others to build/extend from the runtime then too.

Reply:
:100: This can be definitely addressed in Catalyst Runtime. Let's tackle this in a follow-up work as it requires changes upstream. It's good to have the first version of this support working before refining the code base in both Catalyst and Lightning.